### PR TITLE
[WEB-836] [WEB-1011] [WEB-991] [WEB-1133] new clinic admin interface basics

### PR DESCRIPTION
**Pull request by krystophv**
This should take care of [WEB-836] [WEB-1011] [WEB-991] and [WEB-1133]. 
Includes the clinic administration landing page, modifications to clinic account creation, clinician inviting and deleting of invites, deleting and editing clinicians and all associated API linkages.

Pairs with https://github.com/tidepool-org/platform-client/pull/120

[WEB-836]: https://tidepool.atlassian.net/browse/WEB-836
[WEB-1011]: https://tidepool.atlassian.net/browse/WEB-1011
[WEB-991]: https://tidepool.atlassian.net/browse/WEB-991
[WEB-1133]: https://tidepool.atlassian.net/browse/WEB-1133

### CODE REVIEW

**Review by cbwebdevelopment — CHANGES_REQUESTED** [review]
Great job on this!  I do have a bunch of change suggestions, but they almost all fall into one of two categories:

1. Missing translatability wrappers on text
2. Super nit-picky formatting suggestions for consistency (can be ignored)

Aside from this, there were just a couple of instances (captured in comments) where I think we need to respond less optimistically to async requests.

Really a stellar job on this.  Incredibly thorough, and brought to my attention some nice new patterns for handling redux state and route dispatches.  Love it 💯 

I can't say I enjoyed looking over that many lines of test code, but at least I didn't have to write them ;P

File: app/components/elements/Button.js [hunk 1]
```diff
@@ -87,6 +87,7 @@ Button.propTypes = {
     'paginationLight',
     'filter',
     'chip',
+    'danger',
```
[the remark this answers is not in the document]
Comment (krystophv) [reply app/components/elements/Button.js:90]: Ah yeah, didn't catch that in the merge.

File: app/pages/clinicianedit/clinicianedit.js [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    dispatch(push('/clinic-admin'));
+  }
+
+  const clinicAdminDesc = (
+    <>
+      <Title>Clinic Admin</Title>
+      <Body1>
+        Clinic administrators have full read and edit access to access. Clinic
+        administrators have full read and edit access to access{' '}
+      </Body1>
```
[the remark this answers is not in the document]
Comment (krystophv) [reply app/pages/clinicianedit/clinicianedit.js:140]: I honestly don't remember re: the spaces. I'll remove them.

File: app/pages/clinicadmin/clinicadmin.js [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+          variant="textPrimary"
+          onClick={() => handleEdit(userid)}
+        >
+          Edit
+        </Button>
+      );
+    }
+  };
+
+  const renderMore = ({ ...props }) => {
```
[the remark this answers is not in the document]
Comment (krystophv) [reply app/pages/clinicadmin/clinicadmin.js:323]: I don't believe so. I'll just switch it out.

File: app/pages/clinicianedit/clinicianedit.js [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+    trackMetric('Clinic - Edit clinician');
+    dispatch(
+      actions.async.updateClinician(
+        api,
+        selectedClinic,
+        updatedClinician.id,
+        updatedClinician
+      )
+    );
+    dispatch(push('/clinic-admin'));
```
[the remark this answers is not in the document]
Comment (krystophv) [reply app/pages/clinicianedit/clinicianedit.js:131]: Yeah, that's a good call - I'll move the updateClinician toast tracking and redirect on success.

File: app/redux/actions/async.js [hunk 5]
```diff
@@ -1659,32 +1657,23 @@ export function getAllClinics(api, options = {}, cb = _.noop) {
...
-            dispatch(sync.addClinicianToClinicFailure(
-              createActionError(ErrorMessages.ERR_ADDING_CLINICIAN_TO_CLINIC, err), err
-            ));
-          } else {
-            dispatch(sync.addClinicianToClinicSuccess(clinician, clinic.id));
-            console.log('redirect to clinic management interface');
-            // redirect to clinic management interface
-          }
-        })
+        dispatch(push('/clinic-admin'));
```
[the remark this answers is not in the document]
Comment (krystophv) [reply app/redux/actions/async.js:1676]: I'd generally agree, but we have a somewhat established pattern of doing the redirects in actions, especially for key actions in signup/setup flows.

[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply app/redux/actions/async.js:1676]: Ok. Let's leave this as-is, then.

I've looked over all the other changes you've pushed.  This seems to be the last item, so if you re-request a review I'm good to approve.

**Review by cbwebdevelopment — APPROVED** [review]
Great work on this :)
Approved.

**Pull request merged**